### ISSUE TITLE
Remove web dependency

### ISSUE THREAD

**Issue by RPM1984** [danesparza/net-bugsnag#14]
Hey there,

So i was originally using the official library for BugSnag, but ran into an issue regarding OWIN hosting. See [here](https://github.com/bugsnag/bugsnag-dotnet/issues/50). 

I'm looking for alternatives, basically i want a library that can just do the reporting ad-hoc using their Rest API, and not try and hook up automagically to web events.

It seems this library still has a similar approach, in that it takes a dependency on `System.Web` and tries to wire up events. See [here](https://github.com/danesparza/net-bugsnag/blob/master/Bugsnag.Library/BugSnag.cs)

Most of what i need done is in this library. Would you accept a PR that basically allows the consumer of the library to register for web events? 

Basically, i should be able to do this:

```
var bugSnag = new BugSnagClient();
bugSnag.Notify(something) // ad hoc notify
```

The ctor does nothing...sets the api key locally, that's about it. The `Notify` method does the REST call to the notify API.

Then you could have another Nuget package `net-bugsnag.web` which takes a dependency on above, and has something like:

```
var bugSnag = new BugSnagWebClient() // wires up to web events
bugSnag.Notify("something"); // ad hoc notify
```

That way we get the best of both worlds.

Thoughts?
